service/status-report/main.go

```diff
@@ -118,29 +126,63 @@ func main() {
 		}
 	}
 
-	var csiClusterConfigEntry = new(csi.ClusterConfigEntry)
+	if err := updateCSIConfig(ctx, cl, providerClient, storageClient, operatorNamespace); err != nil {
+		klog.Exitf("Failed to update csi config: %v", err)
+	}
+}
+
+func updateCSIConfig(ctx context.Context,
+	cl client.Client,
+	providerClient *providerclient.OCSProviderClient,
+	storageClient *v1alpha1.StorageClient,
+	operatorNamespace string) error {
 	scResponse, err := providerClient.GetStorageConfig(ctx, storageClient.Status.ConsumerID)
 	if err != nil {
-		klog.Exitf("Failed to get StorageConfig of storageClient %v: %v", storageClient.Status.ConsumerID, err)
+		return fmt.Errorf("failed to get StorageConfig of storageClient %v: %v", storageClient.Status.ConsumerID, err)
 	}
 	for _, eResource := range scResponse.ExternalResource {
 		if eResource.Kind == "ConfigMap" && eResource.Name == "rook-ceph-mon-endpoints" {
 			monitorIps, err := csi.ExtractMonitor(eResource.Data)
 			if err != nil {
-				klog.Exitf("Failed to extract monitor data for storageClient %v: %v", storageClient.Status.ConsumerID, err)
+				return fmt.Errorf("failed to extract monitor data for storageClient %v: %v", storageClient.Status.ConsumerID, err)
 			}
+			var csiClusterConfigEntry = new(csi.ClusterConfigEntry)
 			csiClusterConfigEntry.Monitors = append(csiClusterConfigEntry.Monitors, monitorIps...)
+			cc := csi.ClusterConfig{
+				Client:    cl,
+				Namespace: operatorNamespace,
+				Ctx:       ctx,
+			}
+			err = cc.UpdateMonConfigMap("", storageClient.Status.ConsumerID, csiClusterConfigEntry)
+			if err != nil {
+				return fmt.Errorf("failed to update mon configmap for storageClient %v: %v", storageClient.Status.ConsumerID, err)
+			}
+		} else if eResource.Kind == "CephCluster" && utils.DelegateCSI {
+			desiredCephClusterSpec := &csiopv1a1.CephClusterSpec{}
+			if err := json.Unmarshal(eResource.Data, &desiredCephClusterSpec); err != nil {
+				return fmt.Errorf("failed to unmarshall cephClusterSpec: %v", err)
+			}
+
+			cephCluster := &csiopv1a1.CephCluster{}
+			cephCluster.Name = storageClient.Name
+			cephCluster.Namespace = operatorNamespace
+			if err := cl.Get(ctx, client.ObjectKeyFromObject(cephCluster), cephCluster); err != nil {
+				return fmt.Errorf("failed to get csi cephcluster resource: %v", err)
+			}
+
+			if !reflect.DeepEqual(cephCluster.Spec, desiredCephClusterSpec) {
+				cephClusterCopy := &csiopv1a1.CephCluster{}
+				cephCluster.ObjectMeta.DeepCopyInto(&cephClusterCopy.ObjectMeta)
+				desiredCephClusterSpec.DeepCopyInto(&cephClusterCopy.Spec)
+				// patch is being used to ensure spec is accurate at this point of time even if there are reconciles happening which change resourceversion
+				if err := cl.Patch(ctx, cephCluster, client.MergeFrom(cephClusterCopy)); err != nil {
+					return fmt.Errorf("failed to patch csi cephcluster spec: %v", err)
+				}
+			}
```

Comment:
This need to be handled by the operator not the reporter

Reply:
will revisit after https://github.com/red-hat-storage/ocs-client-operator/pull/175#discussion_r1692578565

Reply:
https://github.com/red-hat-storage/ocs-client-operator/pull/175#discussion_r1706417411